### DISCUSSION TITLE
ESP8266 Flash Size: 4MB (FS:3.5MB OTA:~256KB)

### DISCUSSION THREAD

**Discussion by LouisGarry** [esp8266/Arduino#8631] [Q&A] [answered]
Hello masters, how to make Flash Size: 4MB (FS:3.5MB OTA:~256KB) ?
Thanks.

**Comment by d-a-v** [comment]
Without any guaranteed result, you can try and play:

- Find [`boards.txt.py`](https://github.com/esp8266/Arduino/blob/a2982f96a49f40197a56207d6a6a0a4a34fcf90a/tools/boards.txt.py#L1536L1539) in your filesystem
- Edit it and add `f4m.update( flash_map(  4*1024,  nnn))` (`nnn`>3072?)
- [Regenerate configuration](https://arduino-esp8266.readthedocs.io/en/latest/faq/a05-board-generator.html?highlight=boards.txt.py#how-can-i-run-the-script)
- Try

**Reply by LouisGarry** [reply]
Thank you for your replay, I'm using Arduino 1.8.19, and can't find the boards.txt.py file, haha.

**Reply by LouisGarry** [reply]
I want to upload a file size of 3.00 MB (3,153,920 bytes) is it possible?

**Comment by d-a-v** [comment, marked as answer]
> Thank you for your replay, I'm using Arduino 1.8.19, and can't find the boards.txt.py file, haha.

It is [there](https://support.arduino.cc/hc/en-us/articles/360018448279-Open-the-Arduino15-folder).

> I want to upload a file size of 3.00 MB (3,153,920 bytes) is it possible?

If your file is an html file, then you can compress it with `gzip -9` and call it `.html.gz` instead of `.html`.

Otherwise, you can try again and find the above file (`boards.txt.py`) and have a try.
